index.js

```diff
@@ -142,12 +142,39 @@ class HdKeyring extends EventEmitter {
     return Promise.resolve(sig)
   }
 
-  // personal_signTypedData, signs data along with the schema
-  signTypedData (withAccount, typedData) {
-    const wallet = this._getWalletForAccount(withAccount)
-    const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
-    const signature = sigUtil.signTypedData(privKey, { data: typedData })
-    return Promise.resolve(signature)
+  signTypedData (address, typedData, opts = { version: 'V1' }) {
+    if (!address) {
+      throw new Error('Address must be specified')
+    }
+
+    const wallet = this._getWalletForAccount(address, opts)
+    const privateKey = ethUtil.toBuffer(wallet.getPrivateKey())
+
+    switch (opts.version) {
+      case 'V1':
+        return this.signTypedData_v1(typedData, privateKey)
+      case 'V3':
+        return this.signTypedData_v3(typedData, privateKey)
+      case 'V4':
+        return this.signTypedData_v4(typedData, privateKey)
+      default:
+        return this.signTypedData_v1(typedData, privateKey)
+    }
+  }
+
+  signTypedData_v1 (typedData, privateKey) {
```

Comment:
are these methods supposed to be called by external modules? if not i'd prefer they live in the private methods section.

Reply:
they are not, i will update